server/app/services/program/predicate/PredicateGenerator.java

```diff
@@ -27,10 +28,14 @@
 /** Creates {@link PredicateDefinition}s from form inputs. */
 public final class PredicateGenerator {
 
-  private static final Pattern SINGLE_PREDICATE_VALUE_FORM_KEY_PATTERN =
+  private static final Pattern LEGACY_SINGLE_PREDICATE_VALUE_FORM_KEY_PATTERN =
       Pattern.compile("^group-(\\d+)-question-(\\d+)-predicateValue$");
-  private static final Pattern MULTI_PREDICATE_VALUE_FORM_KEY_PATTERN =
+  private static final Pattern LEGACY_MULTI_PREDICATE_VALUE_FORM_KEY_PATTERN =
       Pattern.compile("^group-(\\d+)-question-(\\d+)-predicateValues\\[\\d+\\]$");
+  private static final Pattern SINGLE_PREDICATE_VALUE_FORM_KEY_PATTERN =
+      Pattern.compile("^condition-(\\d+)-subcondition-(\\d+)-value$");
+  private static final Pattern MULTI_PREDICATE_VALUE_FORM_KEY_PATTERN =
+      Pattern.compile("^condition-(\\d+)-subcondition-(\\d+)-values\\[\\d+\\]$");
```

Comment:
So if I'm understanding correctly, the use-case for the MULTI_PREDICATE_VALUE_FORM_KEY_PATTERN is for checkboxes or similar, where there are many values entered?